tools/pythonpkg/tests/fast/spark/test_spark_union_by_name.py

```diff
@@ -32,10 +17,34 @@ def df2(spark):
     yield dataframe
 
 
-# https://sparkbyexamples.com/pyspark/pyspark-unionbyname/
-@pytest.mark.skip(reason="union_by_name is not supported in the Relation API yet")
+@pytest.fixture
+def df3(spark):
+    data2 = [
+        (1,),
+    ]
+    dataframe = spark.createDataFrame(data=data2, schema=["extra_col"])
+    yield dataframe
+
+
 class TestDataFrameUnion(object):
     def test_union_by_name(self, df1, df2):
         rel = df1.unionByName(df2)
-        res = rel.collect()
-        print(res)
+
+        assert rel.columns == ["name", "id"]
+        assert rel.count() == 8
+
+        assert rel.select("name").distinct().count() == 6
+        assert rel.select("id").distinct().count() == 5
+
+    def test_error_union_by_name(self, df1, df3):
+        with pytest.raises(ValueError):
```

Comment:
please use `match: ...` to check for an explicit error, ValueError is too generic